### PR TITLE
Release Pipeline

### PR DESCRIPTION
**Pull request by nirajpathak13**
This PR is to automate the uploading of artifacts when a new release is cut. Once a release is cut, the binaries will be built along with checksums and uploaded to the install/ S3 bucket for download.chia.net.

Here is a video of the pipeline in action:
https://drive.google.com/file/d/18wHIH_g5sd1xCTH_VXw6djiaac9gGzUQ/view?usp=sharing

### CODE REVIEW

**Comment by nirajpathak13** [comment]
This PR is to automate the uploading of artifacts when a new release is cut. Once a release is cut, the binaries will be built along with checksums and uploaded to the install/ S3 bucket for download.chia.net.

Here is a video of the pipeline in action:
https://drive.google.com/file/d/18wHIH_g5sd1xCTH_VXw6djiaac9gGzUQ/view?usp=sharing

**Review by justinengland — DISMISSED** [review]

File: azure-pipelines.yml [hunk 1]
```diff
@@ -48,7 +49,10 @@ steps:
       python -m pip install --upgrade pip
       pip install wheel pep517 setuptools_scm
       node -v
-      pip install -i https://pypi.chia.net/simple/ miniupnpc==2.2.2
+      pip install -i https://pypi.chia.net/simple/ miniupnpc==2.1
```
Comment (hoffmang9) [inline azure-pipelines.yml:52]: Careful - this is a regression from my changes over the weekend to miniupnpc to 2.2.2

**Review by hoffmang9 — CHANGES_REQUESTED** [review]
With that one change this looks good otherwise.

**Review by hoffmang9 — DISMISSED** [review]
AOK

File: .github/workflows/build-macos-installer.yml [hunk 2]
```diff
@@ -108,8 +108,42 @@ jobs:
...
+        aws-region: us-west-2
+
+    - name: Create Checksums
+      run: |
+          ls
+          shasum -a 256 ${{ github.workspace }}/build_scripts/final_installer/Chia-${{ steps.version_number.outputs.CHIA_INSTALLER_VERSION }}.dmg > ${{ github.workspace }}/build_scripts/final_installer/Chia-${{ steps.version_number.outputs.CHIA_INSTALLER_VERSION }}.dmg.checksum
+
+    - name: Upload to s3
+      run: |
+          aws s3 cp ${{ github.workspace }}/build_scripts/final_installer/Chia-${{ steps.version_number.outputs.CHIA_INSTALLER_VERSION }}.dmg s3://download-chia-net/builds/
```
Comment (cmmarslender) [inline .github/workflows/build-macos-installer.yml:142]: Is this going to be the same filename as the builds generated in Azure?

Comment (cmmarslender) [inline .github/workflows/build-macos-installer.yml:142]: Is this going to be the same filename as the builds generated in Azure? I think we're going to be fighting over which Mac job (GHA or Azure) gets the actual spot in S3 depending on which finishes last

**Review by cmmarslender — CHANGES_REQUESTED** [review]

File: .github/workflows/build-macos-installer.yml (hunk 2 above)
Comment (hoffmang9) [reply .github/workflows/build-macos-installer.yml:142]: GHA is invariably last and that's bad as we want to ship the Azure/Mojave build.

**Review by cmmarslender — APPROVED** [review]

**Pull request merged**